PetrolCar.hpp

```diff
@@ -1,16 +1,18 @@
+#pragma once
+#include "Car.hpp"
 #include "PetrolEngine.hpp"
-
-class PetrolCar
-{
+#include <memory>
+class PetrolCar : virtual public Car {
 public:
-    PetrolCar(PetrolEngine* engine);
-    ~PetrolCar();
-    void turnLeft();
-    void turnRight();
-    void brake();
-    void accelerate(int speed);
-    void refuel();
+  PetrolCar(std::unique_ptr<PetrolEngine> engine);
+  ~PetrolCar();
 
-    PetrolEngine* engine_;
-};
+  void restore() override;
+  void accelerate(int a) override;
+  PetrolEngine *getPetrolEngineData() const;
+  std::unique_ptr<PetrolEngine> pullOutPetrolEngine();
```

Comment:
where is putInPetrolEngine function? It is not symetrical with ElectricCar interface.